appcenter-data/Data.js

```diff
@@ -19,22 +19,62 @@ const DefaultPartitions = {
 const Data = {
     TimeToLive,
     DefaultPartitions,
-    read(documentId, partition, readOptions) {
-        if (readOptions === undefined) {
-            readOptions = new Data.ReadOptions(TimeToLive.DEFAULT);
-        }
-        return AppCenterReactNativeData.read(documentId, partition, readOptions).then((result) => {
-            // Create a new `Date` object from timestamp as milliseconds
-            result.lastUpdatedDate = new Date(result.lastUpdatedDate);
-            return result;
-        });
-    }
+    read,
+    create
 };
 
+/**
+ * Read a document.
+ *
+ * @param {string} documentId - The CosmosDB document id.
+ * @param {string} partition - The CosmosDB partition key.
+ * @param {object} readOptions - Cache read options when the operation is done offline.
+ * @return {Promise} Future asynchronous operation with result being the document with metadata.
+ * If the operation fails, the error can be checked by reading the error object.
+ */
+function read(documentId, partition, readOptions) {
+    if (readOptions === undefined) {
+        readOptions = new Data.ReadOptions(TimeToLive.DEFAULT);
+    }
+    return AppCenterReactNativeData.read(documentId, partition, readOptions).then(convertTimestampToDate);
+}
+
+/**
+ * Create a document.
+ *
+ * @param {string} documentId - The CosmosDB document id.
+ * @param {object} document - The document.
+ * @param {string} partition - The CosmosDB partition key.
+ * @param {object} writeOptions - Cache write options when the operation is done offline.
+ * @return {Promise} Future asynchronous operation with result being the document with metadata.
+ * If the operation fails, the error can be checked by reading the error object.
```

Comment:
no it throws an exception on Javascript.

Reply:
> If the operation fails, an exception is thrown and the promise is rejected. The exception can be viewed by reading the error object.

How does that sound?

Reply:
If the operation fails, the promise is rejected with an exception containing the details of the error.

Reply:
Changed